tests/backup/backup_test.go

```diff
@@ -4600,6 +4600,193 @@ var _ = Describe("{RestoreEncryptedAndNonEncryptedBackups}", func() {
+			log.InfoD("Creating bucket, cloud credentials and backup location for testuser1 and testuser2")
+			for _, provider := range providers {
+				BucketName = fmt.Sprintf("%s-%v", provider, buckets[0])
+				cloudCredUID = uuid.New()
+				CloudCredUidList = append(CloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				backup.AddRoleToUser(users[0], backup.InfrastructureOwner, "adding admin role for testuser1")
+				backup.AddRoleToUser(users[1], backup.InfrastructureOwner, "adding admin role for testuser2")
+				UserContextBackupLocation = users[0]
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				time.Sleep(time.Minute * 1)
+				BackupLocationNameUser1 = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				CreateBackupLocation(provider, BackupLocationNameUser1, backupLocationUID, credName, cloudCredUID, BucketName, orgID, "")
+				log.InfoD("Created Backup Location with name - %s", BackupLocationNameUser1)
+				UserContextBackupLocation = users[1]
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				time.Sleep(time.Minute * 1)
+				BackupLocationNameUser2 = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				CreateBackupLocation(provider, BackupLocationNameUser2, backupLocationUID, credName, cloudCredUID, BucketName, orgID, "")
+
+			}
+		})
+		Step("Register source and destination cluster for backup on testuser1", func() {
+			log.InfoD("Registering Source and Destination clusters as user1 and verifying the status")
+			ctx, err := backup.GetNonAdminCtx(users[0], "Password1")
+			log.FailOnError(err, "Fetching testuser1 ctx")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			time.Sleep(time.Minute * 1)
+			portworx.UserContextRegisterCluster = users[0]
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+		})
+
+		Step("Taking backup of applications as testuser1", func() {
+			log.InfoD("Taking backup of applications as user1")
+			var backupName string = "test1"
+			backupNames = append(backupNames, backupName)
+			ctx, err := backup.GetNonAdminCtx(users[0], "Password1")
+			log.FailOnError(err, "Fetching testuser ctx")
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationNameUser1, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, "Verifying backup creation")
+		})
+
+		Step("Register source and destination cluster for backup on testuser2", func() {
+			log.InfoD("Registering Source and Destination clusters as user2 and verifying the status")
+			ctx, err := backup.GetNonAdminCtx(users[1], "Password1")
+			log.FailOnError(err, "Fetching testuser ctx")
```

Comment:
Update the string from testuser to testuser2. So that will get to know for which user its fetching